src/super_context.ml

```diff
@@ -636,40 +836,32 @@ module Action = struct
       ; ddeps     = String.Map.empty
       }
     in
-    let expand var syntax_version =
+    let targets loc name =
+      let var =
+        match name with
+        | "@" -> sprintf "${%s}" name
+        | "targets" -> sprintf "%%{%s}" name
+        | _ -> assert false
+      in
+      match targets_written_by_user with
+      | Infer -> Loc.fail loc "You cannot use %s with inferred rules." var
+      | Alias -> Loc.fail loc "You cannot use %s in aliases." var
+      | Static l -> Some (Value.L.dirs l) (* XXX hack to signal no dep *)
```

Comment:
The `targets` function is used only once, I think it'll be clearer if it's inlined. It took me a while to understand why we needed a hack for instance.